lib/streamlit/hello.py

```diff
@@ -14,57 +14,243 @@
-        take a look at our
-        [tutorials and documentation](https://streamlit.io/docs) next.
-        And feel free to ask us questions any time at
-        [help@streamlit.io](mailto:help@streamlit.io).
+text text text text text text text text text text text text text text text text
+text text text text text text text text text text text text text text text text
+text text text text text text text text text text text text text text text text
+text text text text text text text text text text text text text text text text
+text text text text text text text text text text text text text text text text
+text text text text text text text text text text text text text text text text
+"""
+    )
 
-        To celebrate this magic moment, below we're generating a bunch of
-        random numbers in a loop for around 10 seconds. Enjoy!
+
+def mapping_demo():
     """
-    )
+    This demo shows how to use
+    [`st.deck_gl_chart`](https://streamlit.io/docs/api.html#streamlit.deck_gl_chart)
+    to display geospatial data.
+    """
+    import pandas as pd
+    import copy, os
+    from collections import OrderedDict
+
+    @st.cache
+    def from_data_file(filename):
+        GITHUB_DATA = "https://raw.githubusercontent.com/streamlit/streamlit/develop/examples/"
+        return pd.read_json(os.path.join(GITHUB_DATA, "data", filename))
+
+    ALL_LAYERS = {
+        "Bike Rentals": {
+            "type": "HexagonLayer",
+            "data": from_data_file("bike_rental_stats.json"),
+            "radius": 200,
+            "elevationScale": 4,
+            "elevationRange": [0, 1000],
+            "pickable": True,
+            "extruded": True,
+        },
+        "Bart Stop Exits": {
+            "type": "ScatterplotLayer",
+            "data": from_data_file("bart_stop_stats.json"),
+            "radiusScale": 0.05,
+            "getRadius": "exits",
+        },
+        "Bart Stop Names": {
+            "type": "TextLayer",
+            "data": from_data_file("bart_stop_stats.json"),
+            "getText": "name",
+            "getColor": [0, 0, 0, 200],
+            "getSize": 15,
+        },
+        "Outbound Flow": {
+            "type": "ArcLayer",
+            "data": from_data_file("bart_path_stats.json"),
+            "pickable": True,
+            "autoHighlight": True,
+            "getStrokeWidth": 10,
+            "widthScale": 0.0001,
+            "getWidth": "outbound",
+            "widthMinPixels": 3,
+            "widthMaxPixels": 30,
+        }
+    }
+
+    st.sidebar.markdown('### Map Layers')
+    selected_layers = [layer for layer_name, layer in ALL_LAYERS.items()
+        if st.sidebar.checkbox(layer_name, True)]
+    if selected_layers:
+        viewport={"latitude": 37.76, "longitude": -122.4, "zoom": 11, "pitch": 50}
+        st.deck_gl_chart(viewport=viewport, layers=selected_layers)
+    else:
+        st.error("Please choose at least one layer above.")
 
```

Comment:
Style: 2 spaces between top-level functions. (I thought Black would take care of this? Weird...)

Same throughout the file.